src/components/firstProfile/NameProfile.tsx

```diff
@@ -0,0 +1,134 @@
+import Logo from '../logo/Logo'
+import {
+  Box,
+  Button,
+  FormControl,
+  FormHelperText,
+  OutlinedInput,
+  Typography,
+} from '@mui/material'
+import { useState } from 'react'
+import { makeStyles } from 'tss-react/mui'
+import { commonStyles } from 'styles/commonStyles'
+
+const FULLNAME_REGEX = /^[a-zA-Zа-яА-ЯёЁ\s\p{L}]{2,15}$/u
+
+const NameProfile = () => {
+  const { classes } = useStyles()
+  const commonClasses = commonStyles().classes
+  const [fullName, setFullName] = useState('')
+  const [error, setError] = useState<string>('')
+  const [isValid, setIsValid] = useState<boolean>(false)
+
+  const handleSubmit = async (e: { preventDefault: () => void }) => {
+    e.preventDefault()
+    if (fullName.trim() === '') {
+      setError('Please enter your full name')
+    } else {
+      setError('')
+      setIsValid(false)
+      setFullName('')
+    }
+  }
+
+  const handleInputChange = (event: { target: { value: string } }) => {
+    setError('')
+    setFullName(event.target.value)
+    setIsValid(
+      event.target.value.length >= 2 && !FULLNAME_REGEX.test(event.target.value)
+    )
+  }
+
+  return (
+    <Box className={commonClasses.mainBox}>
+      <Logo />
+      <Typography variant="h1" className={commonClasses.title} pt={10}>
+        Let&apos;s get started!
+      </Typography>
+      <Typography
+        variant="body1"
+        className={`${commonClasses.text} ${classes.profileText}`}
+        pt={5}
+      >
+        What&apos;s your name?
+      </Typography>
+      <form onSubmit={handleSubmit}>
```

Comment:
don't need `form` here. Clicking on the button will add data to local storage. The form usually is needed to send data via API